### PR TITLE
hardware profile detection, fsid overwrite

### PR DESCRIPTION
**Pull request by kmroz**
First pass at some basic hardware profile detection.
Added detection of fsid from existing cluster and overwriting cluster.yml per our TODO.

### CODE REVIEW

File: srv/modules/runners/populate.py [hunk 1]
```diff
@@ -905,6 +921,25 @@ def engulf_existing_cluster(**kwargs):
             # Needs a storage profile assigned (which may be different
             # than the proposals deepsea has come up with, depending on
             # how things were deployed)
+            ceph_disks = local.cmd(minion, "cephinspector.get_ceph_disks_yml")
+            if not ceph_disks:
+                log.error("Failed to get list of Ceph OSD disks.")
+                return [ False ]
+
+            for minion, store in ceph_disks.items():
```
Comment (theanalyst) [inline srv/modules/runners/populate.py]: items is redundant here, `for minion,store in ceph_disks` would work as well

Comment (kmroz) [inline srv/modules/runners/populate.py]: Thanks abhi, will update.

Comment (kmroz) [inline srv/modules/runners/populate.py]: @theanalyst - Hmmm... gives me `ValueError: too many values to unpack`
So I'd need to grab my key via `for minion in ceph_disks:` and access `ceph_disks[minion]`.  Will just keep `.items()`.

Comment (theanalyst) [inline srv/modules/runners/populate.py]: ah nvm, I assumed ceph_disks was a dictionary, it seems it isnt

Comment (kmroz) [inline srv/modules/runners/populate.py]: ceph_disks is a dict... I don't think items()/iteritems() etc, are optional when wanting to extract key and value at once.

File: srv/salt/_modules/cephinspector.py [hunk 2]
```diff
@@ -2,6 +2,60 @@
...
+        ceph_disks["ceph"]["storage"]["osds"][partition]["journal"] = _extract_journal_partition(d)
+
+def get_ceph_disks_yml(**kwargs):
+    """
+    Generates yml representation of Ceph filestores on a given node.
+    Returns something like: {"ceph": {"storage": {"osds": {"/dev/foo":
+                                                            {"format": "filestore",
+                                                             "journal": "/dev/bar"}}}}}
+    """
+    ceph_disk_list = Popen("ceph-disk list", stdout=PIPE, stderr=PIPE, shell=True)
```
Comment (tserong) [inline srv/salt/_modules/cephinspector.py]: You might be better off with `ceph-disk list --format=json`. It gives you more information, and should hopefully be easier to machine parse than the human-readable output.

Comment (kmroz) [inline srv/salt/_modules/cephinspector.py]: @tserong - Cool.  Moved to using --format=json.

**Comment by tserong** [comment]
Sorry, I've just generated some conflicts for you by adding a bit of ceph.conf functionality (also I rebased on 0.7.11)

**Pull request merged**